apps/condo/domains/marketplace/access/Invoice.js

```diff
@@ -44,21 +47,35 @@ async function canReadInvoices ({ authentication: { item: user }, context }) {
 
     const permittedOrganizations = await getEmployedOrRelatedOrganizationsByPermissions(context, user, 'canReadInvoices')
 
+    let canReadAsB2BAppServiceUser = false
+    if (user.type === SERVICE) {
+        // service user may make call as billing integration, when acquiring updates payment to "withdrawn".
+        // But we do not able to check access of service user to integration, because invoice does not contain integration
+        // So we do not return the result here
+        canReadAsB2BAppServiceUser = await canReadObjectsAsB2BAppServiceUser(args)
```

Comment:
wtf? If this is for RB, then RB does not have B2BApp

Reply:
Yes, the integration user has no b2b-rights. However the acquiring integration user needs to read invoice data